packages/lab/stories/form-field-next/form-field-next.doc.stories.mdx

```diff
@@ -0,0 +1,160 @@
+- Text contrast is not WCAG AA compliant; an editable field is only disabled if the field is not relevant at the current time.
+
+## Readonly
+
+Form Field can be set as read-only via the prop ‘readOnly’.
+
+<Canvas>
+  <Story id="lab-form-field-next--readonly" />
+</Canvas>
+
+### Guidance
+
+- The prop should be applied to the field, not the control(s) it contains, for expected results.
+- Set the field to read-only if you’re displaying values that are predefined and not editable, e.g., entry that’s populated and fixed by template selection.
+- Adheres to WCAG AA text contrast for legibility; read-only field content is always relevant to the form, but cannot be changed.
+
+## Field Validation
+
+Form Fields can show validation states (warning, success, error) with the prop `validationStatus`. When the field displays a validation state, the Helper Text is replaced by Validation Text.
+
+<Canvas>
+  <Story id="lab-form-field-next--with-validation" />
+</Canvas>
+
+### Guidance
+
+- The prop should be applied to the field, not the control(s) it contains, for expected results.
+- Use the error state to alert the user of a critical issue that’s related to the field entry. This issue, which may jeopardize completion of the task, usually requires action from the user to resolve the error.
+- Display the warning state when you need to alert the user of a potential issue—that won’t prevent the user from continuing with the task, but may cause errors if it’s not addressed.
+- Validation Text should reiterate and reinforce what was stated in the Helper Text, as it won’t be visible. Make the messaging as specific as possible to help the user resolve the issue.
+
+## How to use
+
+### When to use Form Field
+
+- Use Form Field as a wrapper for editable components in a form layout—such as a simple form, filter panel or cash ticket.
+- We highly recommend using the Form Field Label and Form Field Helper Text components within Form Field to provide your form elements with the descriptive label and validation feedback required to ensure ADA compliance.
+- Align the Form Field to the correct Salt layout grid for your density and viewport size. Set the width of the form control to the number of columns that best reflects the length of the control’s expected value. The form control width can be set independently of the Label and Helper Text if required.
+
+# API
+
+```
+import { FormFieldNext } from "@salt-ds/lab";
+```
+
+## Props
+
+<ArgsTable of={FormFieldNext} />
+
+## CSS Class
+
+<CSSClassTable of={FormFieldNext} />
+
+## Characteristics Used
+
+<CharacteristicUsage of={FormFieldNext} />
+
+## Accessibility
+
+All keyboard interactions are supplied by the underlying wrapped components, rather than the Form Field component.
+
+FormField exposes two props which are important for accessibility: labelId and helperTextId
+
+- `helperTextId` used by `aria-describedby` for the form control
+- `labelId` used by `aria-labelledby` for the form control
```

Comment:
maybe we should mention this in the prop description

Reply:
i realised this was actually wrong, and even idProp wasn't used for them - have updated form field to include the id in restProps instead